motionplan/constraint.go

```diff
@@ -493,3 +482,53 @@ func NewOctreeCollisionConstraint(octree *pointcloud.BasicOctree, threshold int,
 	}
 	return constraint
 }
+
+// NewBoundingRegionConstraint will determine if the given list of robot geometries are in collision with the
+// given list of bounding regions.
+func NewBoundingRegionConstraint(robotGeoms, boundingRegions []spatial.Geometry, collisionBufferMM float64) StateConstraint {
+	return func(state *ik.State) bool {
+		internalGeoms, b := obstacleAndBoudingRegionsConstraintHandler(robotGeoms, state, defaultBoundingRegionConstraintDesc)
+		if !b {
+			return b
+		}
+		cg, err := newCollisionGraph(internalGeoms, boundingRegions, nil, true, collisionBufferMM)
+		if err != nil {
+			return false
+		}
+		return len(cg.collisions(collisionBufferMM)) != 0
+	}
+}
+
+func obstacleAndBoudingRegionsConstraintHandler(
+	robotGeoms []spatial.Geometry, state *ik.State, constraintType string,
+) ([]spatial.Geometry, bool) {
+	var internalGeoms []spatial.Geometry
+	b := true
```

Comment:
why can't we just use errors here?  This is kind of confusing and obfuscates the reason for the error

Reply:
+1

Reply:
`obstacleAndBoudingRegionsConstraintHandler` is a helper function that can return whatever you like.

It is `NewBoundingRegionConstraint` which cannot return an error.